src/main/java/net/openhft/chronicle/queue/internal/reader/queueentryreaders/CustomPluginQueueEntryReader.java

```diff
@@ -0,0 +1,36 @@
+package net.openhft.chronicle.queue.internal.reader.queueentryreaders;
+
+import net.openhft.chronicle.queue.ExcerptTailer;
+import net.openhft.chronicle.queue.reader.ChronicleReaderPlugin;
+import net.openhft.chronicle.queue.reader.MessageConsumer;
+import net.openhft.chronicle.queue.reader.ReadResult;
+import net.openhft.chronicle.wire.DocumentContext;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+public final class CustomPluginQueueEntryReader extends AbstractTailerPollingQueueEntryReader {
+
+    private final ChronicleReaderPlugin plugin;
+    private final MessageConsumer consumer;
+    private final ThreadLocal<AtomicInteger> tlMatchCounter = ThreadLocal.withInitial(AtomicInteger::new);
+
+    public CustomPluginQueueEntryReader(ExcerptTailer tailer, Function<ExcerptTailer, DocumentContext> pollMethod, ChronicleReaderPlugin plugin,
+                                        MessageConsumer consumer) {
+        super(tailer, pollMethod);
+        this.plugin = plugin;
+        this.consumer = consumer;
+    }
+
+    @Override
+    protected ReadResult doRead(DocumentContext documentContext) {
+        AtomicInteger matchCounter = tlMatchCounter.get();
+        matchCounter.set(0);
+        plugin.onReadDocument(documentContext, value -> {
+            if (consumer.consume(documentContext.index(), value)) {
+                matchCounter.incrementAndGet();
+            }
+        });
+        return new ReadResult(true, matchCounter.get());
```

Comment:
Creates an object.

Reply:
Do we need to care about garbage for ChronicleReader? isn't it a command line tool for dumping the contents of a queue?

Reply:
That's right. Then this is perfectly ok.

Reply:
Managed to remove the object creation anyway :stuck_out_tongue_winking_eye: